File: frontend/src/api/transaction.ts
import request from './index'

export interface Transaction {
  record_id: string
  amount: number
  category_id?: string
  project_id?: string
  project_name?: string
  person_id?: string
  transaction_time: string
  remark: string
  status: number
  creator_id: string
  create_time: string
  update_time: string
  category?: {
    category_id: string
    name: string
  }
  project?: {
    project_id: string
    name: string
    description: string
  }
  person?: {
    user_id: string
    username: string
    real_name: string
  }
  creator?: {
    user_id: string
    username: string
    real_name: string
  }
  attachments?: Attachment[]
}

export interface Attachment {
  attachment_id: string
  record_id: string
  file_name: string
  file_path: string
  file_size: number
  file_type: string
  upload_time: string
}

export interface CreateTransactionParams {
  amount: number
  category_id?: string
  project_id?: string
  person_id?: string
  transaction_time: string
  remark?: string
  attachment_ids: string[]
}

export interface TransactionListParams {
  page?: number
  page_size?: number
  start_time?: string
  end_time?: string
  category_id?: string
  project_id?: string
  person_id?: string
  type?: 'income' | 'expense' | 'all'
}

// 创建收支记录
export function createTransaction(data: CreateTransactionParams) {
  return request.post('/transactions', data)
}

// 获取收支记录列表
export function getTransactionList(params: TransactionListParams) {
  return request.get('/transactions', { params })
}

// 获取收支记录详情
export function getTransactionDetail(id: string) {
  return request.get(`/transactions/${id}`)
}

// 更新收支记录
export function updateTransaction(id: string, data: Partial<Transaction>) {
  return request.put(`/transactions/${id}`, data)
}

// 删除收支记录
export function deleteTransaction(id: string) {
  return request.delete(`/transactions/${id}`)
}

// 审核通过
export function approveTransaction(id: string, remark?: string) {
  return request.put(`/transactions/${id}/approve`, { remark })
}

// 驳回
export function rejectTransaction(id: string, reason: string) {
  return request.put(`/transactions/${id}/reject`, { reason })
}

File: backend/models/transaction.go
package models

import (
	"time"
)

// Transaction 收支流水模型
type Transaction struct {
	RecordID         string         `gorm:"column:record_id;primaryKey;size:32" json:"record_id"`
	Amount           float64        `gorm:"column:amount;type:decimal(15,2);not null" json:"amount"` // 正数=收入，负数=支出
	CategoryID       *string        `gorm:"column:category_id;size:32" json:"category_id"`
	ProjectID        *string        `gorm:"column:project_id;size:32" json:"project_id"` // 关联项目ID
	PersonID         *string        `gorm:"column:person_id;size:32" json:"person_id"`
	PaymentMethodID  *string        `gorm:"column:payment_method_id;size:32" json:"payment_method_id"` // 支付方式ID
	TransactionTime  time.Time      `gorm:"column:transaction_time;not null" json:"transaction_time"`
	Remark           string         `gorm:"column:remark;size:500" json:"remark"`
	Status           int            `gorm:"column:status;default:0" json:"status"` // 0-待审核，1-已审核，2-已驳回
	CreatorID        string         `gorm:"column:creator_id;size:32;not null" json:"creator_id"`
	IsDeleted        int            `gorm:"column:is_deleted;default:0" json:"is_deleted"` // 0-未删除，1-已删除
	CreateTime       time.Time      `gorm:"column:create_time;autoCreateTime" json:"create_time"`
	UpdateTime       time.Time      `gorm:"column:update_time;autoUpdateTime" json:"update_time"`

	// 关联
	Category      *Category      `gorm:"foreignKey:CategoryID;References:CategoryID;constraint:OnDelete:SET NULL,OnUpdate:CASCADE" json:"category,omitempty"`
	Project       *Project       `gorm:"foreignKey:ProjectID;References:ProjectID;constraint:OnDelete:SET NULL,OnUpdate:CASCADE" json:"project,omitempty"`
	Person        *User          `gorm:"foreignKey:PersonID;References:UserID;constraint:OnDelete:SET NULL,OnUpdate:CASCADE" json:"person,omitempty"`
	PaymentMethod *PaymentMethod `gorm:"foreignKey:PaymentMethodID;References:PaymentMethodID;constraint:OnDelete:SET NULL,OnUpdate:CASCADE" json:"payment_method,omitempty"`
	Creator       *User          `gorm:"foreignKey:CreatorID;References:UserID" json:"creator,omitempty"`
	Attachments   []Attachment   `gorm:"foreignKey:RecordID;References:RecordID;constraint:OnDelete:CASCADE,OnUpdate:CASCADE" json:"attachments,omitempty"`
}

// TableName 指定表名
func (Transaction) TableName() string {
	return "t_transaction"
}
